crates/ethereum/src/lib.rs

```diff
@@ -78,7 +91,7 @@ impl EthereumClient {
         });
         let res: serde_json::Value = self
             .http
-            .post(&self.rpc_url)
```

Comment:
This one is really weird, `post` method was complaining that `Url` does not impl `IntoUrl`...